### PR TITLE
P1456R1 Move-only views

### LINKED ISSUE #3414: [2019-11 LWG Motion 11] P1456R1 Move-only views

### PR DESCRIPTION
**Pull request by zygoloid**
Fixes #3414.

Fixes cplusplus/nbballot#273
Partially addresses cplusplus/nbballot#249
Partially addresses cplusplus/nbballot#266
Fixes cplusplus/nbballot#277

### CODE REVIEW

**Comment by jensmaurer** [comment]
Please augment the commit message to add references to the NB ballot comments mentioned above.

**Comment by johelegp** [comment]
Missing changes to `base`: `take_while_view`, `drop_view`, `drop_while_view` and `elements_view`.

**Comment by cor3ntin** [comment]
@johelegp No, you are right, all views with  a `base` member function should provide a rvale qualified overload - P1862 does that to iterators ( and iterators only ) @CaseyCarter

**Comment by CaseyCarter** [comment]
> Missing changes to base: `take_while_view`, `drop_view`, `drop_while_view` and `elements_view`.

Yes, "Change [range.transform.view], [range.take.view], [range.common.view], and [range.reverse.view] similarly" in P1456 is missing the range adaptors added by P1035. Are we willing to construe P1456 as providing editorial instructions to make similar changes to all such range adaptors, or shall I submit an LWG issue to make this same change to these four that were missed?

**Comment by zygoloid** [comment]
@CaseyCarter I think the "Proposal" section gives us sufficient motivation to consider this an editorial fix to the wording as moved:

> we propose that each such `base()` member be replaced to by two overloads: a `const`-qualified overload that requires the type of the underlying view to model CopyConstructible, and a `&&`-qualified overload that extracts the underlying view from the adaptor

(especially with you as a paper author telling us this was indeed the intent!).

File: papers/nxxxx.md [hunk 1]
```diff
@@ -43,6 +43,25 @@ CWG motion 2: [P1234R5 "Paper name"](http://wg21.link/p1234r5), resolving 3 NB c
...
+
+but the wording changes omitted explicit editing instructions
+to make these changes to the
+`take_while_view`, `drop_view`, `drop_while_view` and `elements_view`
+range adaptors, which were added by [P1035R7](http://wg21.link/p1035r7)
+(2019-07 LWG Motion 23), after R0 of this paper was authored.
+
+Consistent with the proposal in the paper,
+and after consulting the paper authors and the LWG chair,
+the corresponding changes were also applied to the five additional range adaptors.
```
Comment (johelegp) [inline papers/nxxxx.md]:
```suggestion
the corresponding changes were also applied to the four additional range adaptors.
```

Comment (johelegp) [reply papers/nxxxx.md]: Hmm... This can be technically correct and not at the same time.

Comment (CaseyCarter) [reply papers/nxxxx.md]: This is ironically reminiscent of #3321: Never list things, and never count them ;)

Comment (zygoloid) [reply papers/nxxxx.md]: Oops. You'd think I'd be able to count to four. That's embarrassing. =)

Fixed by removing the number entirely. Thanks for catching that.

**Pull request merged**